### PR TITLE
#13 카드 리스트 -> 카드 상세 전환 모션 작업

### PR DESCRIPTION
**Pull request by seunghee63**
- https://github.com/Nexters/talkbbokki-android/issues/13
- 디테일한 에니메이션은 디자인 나오면 적용 해 볼게요 ~~~ !

+ CI 중복으로 돌아가는것도 수정했심당 ~

### CODE REVIEW

**Review by Jiyoung9310 — APPROVED** [review]
애니메이션 마스터 가쥬아~

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailDestination.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.hammer.talkbbokki.presentation.detail
+
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavType
+import androidx.navigation.compose.composable
+import androidx.navigation.navArgument
+import com.hammer.talkbbokki.presentation.navigation.TalkbbokkiNavigationDestination
+
+
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailDestination.kt:9]: 🚫 **[ktlint]** <no-consecutive-blank-lines> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Needless blank line(s)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 2]
```diff
@@ -0,0 +1,143 @@
+package com.hammer.talkbbokki.presentation.detail
+
+import androidx.compose.animation.core.FastOutSlowInEasing
+import androidx.compose.animation.core.animate
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:7]: 🚫 **[ktlint]** <no-wildcard-imports> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wildcard import (cannot be auto-corrected)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+import androidx.compose.material.Text
+import androidx.compose.material3.Button
+import androidx.compose.runtime.*
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:10]: 🚫 **[ktlint]** <no-wildcard-imports> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wildcard import (cannot be auto-corrected)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            .size(192.dp, 240.dp)
+    ) {
+        TestFlipCard(cardFace, onClickToList, id)
+    }
+
+    if (cardFace == CardFace.FRONT) {
+        Button(
+            onClick = {
+                cardFace = CardFace.BACK
+            }, modifier = modifier.wrapContentSize()
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:47]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    var scale by remember { mutableStateOf(1f) }
+    var rotation by remember { mutableStateOf(1f) }
+
+    FlipCard(
+        rotation = rotation,
+        modifier = Modifier
+            .fillMaxWidth()
+            .aspectRatio(.7f)
+            .graphicsLayer(
+                scaleX = scale, scaleY = scale, cameraDistance = 12f
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:65]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:65]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            ),
+        front = {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(Color.Blue),
+                contentAlignment = Alignment.Center,
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:72]: 🚫 **[ktlint]** <trailing-comma-on-call-site> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary trailing comma before ")"

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+            ) {
+                Text(
+                    text = id,
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:75]: 🚫 **[ktlint]** <trailing-comma-on-call-site> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary trailing comma before ")"

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+                )
+            }
+        },
+        back = {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(Color.Blue),
+                contentAlignment = Alignment.Center,
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:84]: 🚫 **[ktlint]** <trailing-comma-on-call-site> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary trailing comma before ")"

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+            ) {
+                Text(
+                    text = "대화주제",
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:87]: 🚫 **[ktlint]** <trailing-comma-on-call-site> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary trailing comma before ")"

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+                )
+            }
+        },
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:90]: 🚫 **[ktlint]** <trailing-comma-on-call-site> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary trailing comma before ")"

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+    )
+
+    if (cardFace == CardFace.FRONT) {
+        LaunchedEffect(key1 = Unit) {
+            coroutineScope {
+                launch {
+                    animate(
+                        initialValue = 1f, targetValue = 1.7f, animationSpec = tween(
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:98]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:98]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline after ","

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:98]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+                            durationMillis = 1000, easing = FastOutSlowInEasing
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:99]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+                        )
+                    ) { value: Float, _: Float ->
+                        scale = value
+                    }
+
+                    animate(
+                        initialValue = 0f, targetValue = 180f, animationSpec = tween(
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:106]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:106]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline after ","

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:106]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+                            durationMillis = 540, easing = FastOutSlowInEasing
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:107]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+                    }
+                }
+            }
+        }
+    } else {
+        LaunchedEffect(key1 = Unit) {
+            coroutineScope {
+                launch {
+                    animate(
+                        initialValue = 180f, targetValue = 0f, animationSpec = tween(
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:120]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:120]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline after ","

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:120]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+                            durationMillis = 540, easing = FastOutSlowInEasing
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:121]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+                        )
+                    ) { value, _ ->
+                        rotation = value
+                    }
+
+                    animate(
+                        initialValue = 1.7f, targetValue = 1f, animationSpec = tween(
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:128]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:128]: 🚫 **[ktlint]** <wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing newline after ","

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:128]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+                            durationMillis = 1000, easing = FastOutSlowInEasing
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailScreen.kt:129]: 🚫 **[ktlint]** <argument-list-wrapping> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Argument should be on a separate line (unless all arguments can fit a single line)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailViewModel.kt [hunk 19]
```diff
@@ -0,0 +1,13 @@
+package com.hammer.talkbbokki.presentation.detail
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailViewModel.kt:1]: 🚫 **[ktlint]** <final-newline> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>File must end with a newline (\n)

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailViewModel.kt [hunk 20]
```diff
@@ -0,0 +1,13 @@
...
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class DetailViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle
+) : ViewModel() {
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailViewModel.kt:11]: 🚫 **[ktlint]** <no-empty-class-body> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unnecessary block ("{}")

File: app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailViewModel.kt [hunk 21]
```diff
@@ -0,0 +1,13 @@
...
+
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/detail/DetailViewModel.kt:12]: 🚫 **[ktlint]** <no-blank-line-before-rbrace> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected blank line(s) before "}"

**Review by seunghee63 — COMMENTED** [review]
Remaining comments which cannot be posted as a review comment to avoid GitHub Rate Limit

<details>
<summary>ktlint</summary>

[app/src/main/java/com/hammer/talkbbokki/presentation/detail/FlipCard.kt|15 col 39|](http://github.com/Nexters/talkbbokki-android/blob/2be2da961d63a93748c700575cb6efd523355b8e/app/src/main/java/com/hammer/talkbbokki/presentation/detail/FlipCard.kt#L15) Unnecessary trailing comma before ")"
[app/src/main/java/com/hammer/talkbbokki/presentation/detail/FlipCard.kt|22 col 14|](http://github.com/Nexters/talkbbokki-android/blob/2be2da961d63a93748c700575cb6efd523355b8e/app/src/main/java/com/hammer/talkbbokki/presentation/detail/FlipCard.kt#L22) Unnecessary trailing comma before ")"
[app/src/main/java/com/hammer/talkbbokki/presentation/detail/FlipCard.kt|36 col 22|](http://github.com/Nexters/talkbbokki-android/blob/2be2da961d63a93748c700575cb6efd523355b8e/app/src/main/java/com/hammer/talkbbokki/presentation/detail/FlipCard.kt#L36) Unnecessary trailing comma before ")"
[app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListDestination.kt|13 col 27|](http://github.com/Nexters/talkbbokki-android/blob/2be2da961d63a93748c700575cb6efd523355b8e/app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListDestination.kt#L13) Unexpected spacing before ":"
[app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListScreen.kt|31 col 26|](http://github.com/Nexters/talkbbokki-android/blob/2be2da961d63a93748c700575cb6efd523355b8e/app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListScreen.kt#L31) Unexpected spacing before ":"
[app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListScreen.kt|40 col 73|](http://github.com/Nexters/talkbbokki-android/blob/2be2da961d63a93748c700575cb6efd523355b8e/app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListScreen.kt#L40) Unexpected spacing before ":"
[app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListScreen.kt|75 col 24|](http://github.com/Nexters/talkbbokki-android/blob/2be2da961d63a93748c700575cb6efd523355b8e/app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListScreen.kt#L75) Missing newline after "{"
</details>

**Pull request merged**